core/src/apps/solana/sign_tx.py

```diff
@@ -167,4 +185,136 @@ def calculate_fee(transaction: Transaction) -> int:
                 unit_price = instruction.lamports
                 is_unit_price_set = True
 
-    return int(base_fee + math.ceil(unit_price * unit_limit / 1000000))
+    priority_fee = math.ceil(unit_price * unit_limit / 1000000)
+    return base_fee, priority_fee
+
+
+def calculate_rent_exemption(transaction: Transaction) -> int:
+    """
+    Returns max rent exemption in lamports.
+
+    To estimate rent exemption from a transaction we need to go over the instructions.
+    When new accounts are created, space must be allocated for them, rent exemption value depends on that space.
+
+    There are a handful of instruction that allocate space:
+    - System program create account instruction (the space data parameter)
+    - System program create account with seed instruction (the space data parameter)
+    - System program allocate instruction (the space data parameter)
+    - System program allocate with seed instruction (the space data parameter)
+    - Associated token account program create instruction (165 bytes for Token, ??? for Token22)
+    - Associated token account program create idempotent instruction (165 bytes for Token, ??? for Token22, might not allocate)
+    """
```

Comment:
I'd also mention as a note that Token program doesn't really create accounts because they all have to be pre-allocated via System Program.